engine/src/main/java/org/hibernate/search/cfg/Environment.java

```diff
@@ -218,6 +218,37 @@
 	 */
 	public static final String INDEX_NAME_JMS_PROPERTY = "HSearchIndexName";
 
+	/**
+	 * Option for setting the locking strategy to be used for locking Lucene directories. Supported values are:
+	 * <ul>
+	 * <li><code>simple</code></li>
+	 * <li><code>native</code></li>
+	 * <li><code>single</code></li>
+	 * <li><code>none</code></li>
+	 * <li>The fully-qualified name of a custom {@link org.hibernate.search.store.LockFactoryProvider} implementation.</li>
+	 * </ul>
+	 * Can be given globally or for specific indexes:
+	 * <ul>
+	 * <li><code>hibernate.search.default.locking_strategy=simple</code></li>
+	 * <li><code>hibernate.search.Books.locking_strategy=org.custom.components.MyLockingFactory</code></li>
+	 * </ul>
+	 */
+	public static final String LOCKING_STRATEGY = "locking_strategy";
```

Comment:
I like that you add these to `Environment`. Not all properties are added there :-(